### PR TITLE
Redesign AveragePool

### PR DESCRIPTION
**Pull request by Hyunseok-Kim0**
### 1. Content and background

I used FastestDet and code below for validity test. This is may not enough, need more test case and the results.
```python
import os
import shutil

import onnxruntime
import tensorflow as tf
import torch
import torchvision
from einops import rearrange
from torch import nn
import numpy as np
from onnx2tf import convert
import onnx_graphsurgeon as gs
import onnx


class dummy_network(nn.Module):

    def __init__(self, kernel=3, pad=1, stride=1):
        super(dummy_network, self).__init__()
        self.kernel = kernel
        self.pad = pad
        self.stride = stride

    def forward(self, x):

        if len(x.shape) == 3:
            result = torch.nn.AvgPool1d(self.kernel, self.stride, self.pad, count_include_pad=False)(x)

        elif len(x.shape) == 4:
            result = torch.nn.AvgPool2d(self.kernel, self.stride, self.pad, count_include_pad=False)(x)

        else:
            result = torch.nn.AvgPool3d(self.kernel, self.stride, self.pad, count_include_pad=False)(x)

        return result


def main():
    os.makedirs("tflite", exist_ok=True)

    dummy_name = "dummy_avgpool"
    onnx_save_path = f"tflite/{dummy_name}.onnx"
    temp_tflite = "tflite/model_float32.tflite"
    tflite_save_path = f"tflite/{dummy_name}.tflite"

    dummy_x = torch.randn(1, 48, 64, 64, dtype=torch.float32)
    np.random.seed(12)
    kernels = [np.random.randint(3, 8) for _ in range(30)]
    strides = [np.random.randint(2, k) for k in kernels]
    paddings = [np.random.randint(0, k / 2) for k in kernels]

    for k, s, p in zip(kernels, strides, paddings):
        print("kernel, stride, padding : ", k, s, p)
        model = dummy_network(kernel=k, stride=s, pad=p)
        model.eval()

        torch.onnx.export(model,
                          args=(dummy_x, ),
                          f=onnx_save_path,
                          input_names=["x"],
                          output_names=["result"],
                          opset_version=11)

        # randomly turn on count_include_pads
        if np.random.rand() > 0.5:
            graph = gs.import_onnx(onnx.load(onnx_save_path))

            for i, gn in enumerate(graph.nodes):
                graph.nodes[i].attrs['count_include_pad'] = 1
            graph.toposort()
            onnx.save(gs.export_onnx(graph), onnx_save_path)

        # get onnx output
        # -----------------------------------------------------------------------------------------------
        onnx_session = onnxruntime.InferenceSession(onnx_save_path, providers=['CPUExecutionProvider'])
        onnx_inputs = dict(x=dummy_x.numpy())
        onnx_output = onnx_session.run(None, onnx_inputs)[0]

        convert(onnx_save_path, output_folder_path="tflite", check_onnx_tf_outputs_elementwise_close_full=True)
        shutil.move(temp_tflite, tflite_save_path)

        # get tflite output
        # -----------------------------------------------------------------------------------------------
        tflite_onnx2tf = tf.lite.Interpreter(model_path=tflite_save_path)
        tflite_onnx2tf.allocate_tensors()

        if len(dummy_x.shape) == 3:
            tflite_onnx2tf.set_tensor(tflite_onnx2tf.get_input_details()[0]['index'],
                                      rearrange(dummy_x.numpy(), "n c h -> n h c"))
        elif len(dummy_x.shape) == 4:
            tflite_onnx2tf.set_tensor(tflite_onnx2tf.get_input_details()[0]['index'],
                                      rearrange(dummy_x.numpy(), "n c h w -> n h w c"))
        elif len(dummy_x.shape) == 5:
            tflite_onnx2tf.set_tensor(tflite_onnx2tf.get_input_details()[0]['index'],
                                      rearrange(dummy_x.numpy(), "n c d h w -> n d h w c"))

        tflite_onnx2tf.invoke()
        tflite_output = [tflite_onnx2tf.get_tensor(i['index']) for i in tflite_onnx2tf.get_output_details()][0]

        if len(dummy_x.shape) == 3:
            tflite_output = rearrange(tflite_output, "n h c -> n c h")
        elif len(dummy_x.shape) == 4:
            tflite_output = rearrange(tflite_output, "n h w c -> n c h w")
        elif len(dummy_x.shape) == 5:
            tflite_output = rearrange(tflite_output, "n d h w c -> n c d h w")

        np.testing.assert_allclose(tflite_output, onnx_output, rtol=0.1, atol=1e-3)
        print("convert done")

    return


if __name__ == "__main__":
    main()

```

### 2. Summary of corrections
I added new function to calculate how much padding is needed for tensorflow `SAME`. Because tensorflow append padding not evenly if the odd number padding is needed, I also added new logic to reproduce ONNX pooling action in tensorflow. 

Now, additional `Pad` layer is only added if needed. Also, extra `StridedSlice` layer, `Mul` layer and `Concatenation` layer is added to calculate same output of ONNX if needed.

### 4. Issue number (only if there is a related issue)
#124

### CODE REVIEW

**Comment by PINTO0309** [comment]
Amazing.

**Comment by Hyunseok-Kim0** [comment]
I'm worried about the lack of additional test cases, which makes it difficult to be sure that this PR is complete.
Anyway, it seems that `MaxPool` also needs to be modified, and it can be easily modified with slight variation in this PR. If this PR and additional tests are done without any problems, I will continue with the work.

**Comment by Hyunseok-Kim0** [comment]
It looks the test is failed while converting `models/imageclassifier.onnx`. Can I get the file for debugging?

**Comment by PINTO0309** [comment]
All regression test ONNX files are posted here.
https://github.com/PINTO0309/onnx2tf/releases/tag/1.1.28

**Comment by Hyunseok-Kim0** [comment]
Couldn't find error until now, ready to be merged.
